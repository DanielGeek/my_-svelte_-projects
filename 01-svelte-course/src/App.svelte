<script>
	// import Counter from './lib/Counter.svelte';
	// import Button from './lib/Button.svelte';
	// import FaAngellist from 'svelte-icons/fa/FaAngellist.svelte';
	// import FaAllergies from 'svelte-icons/fa/FaAllergies.svelte';

	import TodoList from './lib/TodoList.svelte';
	import { v4 as uuid } from 'uuid';

	let todos = [
		{
			id: uuid(),
			title: 'Todo 1',
			completed: true,
		},
		{
			id: uuid(),
			title: 'Todo 2',
			completed: false,
		},
		{
			id: uuid(),
			title: 'Todo 3',
			completed: true,
		},
	];
	$: console.log(todos);
	// const props = {
	//   maxCount: 10,
	//   initialCount: 5
	// };
</script>

<h2>{todos.length} Todos</h2>
<TodoList bind:todos />

<button
	on:click={() => {
		todos = [];
	}}
	>Update todos
</button>

<!-- <Button
	on:click|once={(event) => {
		alert(true);
	}}
	let:isLeftHovered
	size="small"
	shadow
	disabled
>
	<div style:width="20px" slot="leftContent">
		{#if isLeftHovered}
			<FaAngellist />
		{:else}
			<FaAllergies />
		{/if}
	</div>
	Button Text
</Button> -->

<!-- <Counter {...props} x="y" /> -->
<style>
</style>
