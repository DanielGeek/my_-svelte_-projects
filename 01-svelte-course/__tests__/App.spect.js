import { render } from '@testing-library/svelte';
import App from '../src/App.svelte';

describe('App', () => {
	it('Snapshot', () => {
		const { component } = render(App);
		expect(component).toMatchSnapshot();
	});

	it('component should be defined', () => {
		const { component } = render(App);
		expect(component).toBeDefined();
	});
});
