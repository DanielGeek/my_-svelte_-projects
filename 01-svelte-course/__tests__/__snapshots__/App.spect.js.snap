// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Snapshot 1`] = `
App {
  "$$": {
    "after_update": [],
    "before_update": [],
    "bound": {},
    "callbacks": {},
    "context": Map {},
    "ctx": [],
    "dirty": [
      -1,
    ],
    "fragment": {
      "c": [Function],
      "d": [Function],
      "i": [Function],
      "l": [Function],
      "m": [Function],
      "o": [Function],
      "p": [Function],
    },
    "not_equal": [Function],
    "on_destroy": [
      [Function],
    ],
    "on_disconnect": [],
    "on_mount": [],
    "props": {},
    "root": <div />,
    "skip_bound": false,
    "update": [Function],
  },
  "$capture_state": [Function],
}
`;
